src/main/java/org/jenkinsci/plugins/docker/workflow/declarative/AbstractDockerAgent.java

```diff
@@ -123,4 +125,18 @@ public boolean reuseRootAgent(Map<String, DeclarativeOption> options) {
         return options.get(ContainerPerStage.SYMBOL) != null;
     }
 
+    /**
+     * AbstractDockerPipelineScript.groovy is a superclass of the Groovy scripts for subclasses of
+     * {@link AbstractDockerAgent}, but does not have any direct equivalent Java class, so we just allow it here.
+     */
+    @Extension
+    public static class ChangelogConditionalScriptAllowlist extends GroovySourceFileAllowlist {
+        private final String scriptUrl = AbstractDockerAgent.class.getResource("AbstractDockerPipelineScript.groovy").toString();
```

Comment:
Do I get it right, that if `AbstractDockerPipelineScript` is allowed, then all classes that extend it are also allowed?

Reply:
No, they have to be allowed individually. However, the concrete classes are all used by subclasses of `WithScriptDescribable`, which are handled automatically by https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/529. See specifically `WithScriptAllowist`: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/529#discussion_r875061290.